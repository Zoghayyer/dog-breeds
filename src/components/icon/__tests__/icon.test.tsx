import React from 'react';
import { shallow } from 'enzyme';
import Icon, { IconProps } from '../icon';

describe('(Component) Icon', () => {
  let component;
  const defaultProps: IconProps = {
    alt: 'My fake alt text',
    className: '',
    icon: 'redHeartIcon',
  };

  const shallowIcon = (extendedProps: Partial<IconProps> = {}) => {
    component = shallow(<Icon {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowIcon());

  it('renders without crashing', () => {
    shallow(<Icon {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  it('renders the correct icon', () => {
    const wrapper = shallow(<Icon {...defaultProps} />);
    expect(wrapper.prop('alt')).toBe(defaultProps.alt);
  });

  it('calls onClick on icon click', () => {
    const onClickSpy = jest.fn();
    shallowIcon({ onClick: onClickSpy });

    component.find('img').simulate('click');
    expect(onClickSpy).toHaveBeenCalledTimes(1);
  });
});
