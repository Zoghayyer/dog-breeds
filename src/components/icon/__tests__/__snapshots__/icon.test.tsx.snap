// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Icon renders without crashing 1`] = `
<img
  alt="My fake alt text"
  className="icon "
  src="test-file-stub"
/>
`;
